src/components/TextInput/TextInput.tsx

```diff
@@ -15,29 +17,38 @@ export type TextInputProps = NativeAttributes<'input'> & {
 
   /** Whether the input is disabled or not */
   disabled?: boolean;
+
+  /** The icon present on the input  */
+  icon?: IconProps['type'];
 };
 
 const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(function TextInput(
-  { label, invalid, required, disabled, id, hidden, name, value, ...rest },
+  { label, invalid, required, disabled, id, hidden, name, icon, value, ...rest },
   ref
 ) {
   const identifier = id || name || slugify(label);
 
   return (
     <InputControl invalid={invalid} disabled={disabled} required={required} hidden={hidden}>
-      <InputElement
-        as="input"
-        type="text"
-        id={identifier}
-        name={name}
-        value={value}
-        truncated
-        {...rest}
-        ref={ref}
-      />
-      <InputLabel raised={!!value} htmlFor={identifier}>
-        {label}
-      </InputLabel>
+      <Flex align="center" position="relative">
+        <InputElement
+          as="input"
+          type="text"
+          id={identifier}
+          name={name}
+          value={value}
+          truncated
+          pr={icon && 40}
+          position="relative"
```

Comment:
not use if `position: relative` is needed

Reply:
Position relative is needed as `z-index` is used in order to stack up and send the icon to the "background".